File: engine/engine-model/src/main/scala/tools/packet/cashshop/operation/ShowGifts.scala
package tools.packet.cashshop.operation

import net.opcodes.SendOpcode
import tools.packet.{Gift, PacketInput}

class ShowGifts(private var _gifts: java.util.List[Gift]) extends PacketInput {
  def gifts: java.util.List[Gift] = _gifts

  override def opcode(): SendOpcode = SendOpcode.CASH_SHOP_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/cashshop/operation/ShowGifts.java
package tools.packet.cashshop.operation;

import java.util.List;

import net.opcodes.SendOpcode;
import tools.packet.Gift;
import tools.packet.PacketInput;

public record ShowGifts(List<Gift> gifts) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.CASH_SHOP_OPERATION;
   }
}